### PR TITLE
This is just to talk about Dependency Injection

### PR DESCRIPTION
**Pull request by daniel-dgi**

### CODE REVIEW

**Comment by nigelgbanks** [comment]
Just thought I should point out that there is exactly 1,337 additions.

File: services/CollectionService/src/index.php [hunk 1]
```diff
@@ -22,23 +23,12 @@
   'twig.path' => __DIR__.'/../templates',
 ));
 
-$islandoraResourceServiceProvider = new \Islandora\ResourceService\Provider\ResourceServiceProvider;
```
Comment (DiegoPino) [inline services/CollectionService/src/index.php]: Sorry, some how lost here. How does the collection.controller makes use of the resource routes without having them registered or mounted? Who is mounting them? Thanks!

Comment (daniel-dgi) [inline services/CollectionService/src/index.php]: It's totally non-functional.  Was just trying to make a point about injecting things in and making sure to use PHP's type system for gurantees.  I've yet to pull out the rest of the Resource Service stuff.

Comment (DiegoPino) [inline services/CollectionService/src/index.php]: Oh, right! Well sorry for being so silly. Thanks! I will go inject myself :+1:

Comment (daniel-dgi) [inline services/CollectionService/src/index.php]: LOL, no worries.

**Pull request closed without merging**